src/data/gaussian_2d_data_loader.py

```diff
@@ -2,55 +2,100 @@
 import torch
 from torch.utils.data import Dataset
 import matplotlib
+
+matplotlib.use("Agg")
 import matplotlib.pylab as plt
+import matplotlib.cm as cm
 import numpy as np
 from helpers.pytorch_helpers import to_pytorch_variable
 from helpers.configuration_container import ConfigurationContainer
 from data.data_loader import DataLoader
 from math import sqrt
 
-matplotlib.use("Agg")
-
-
 N_RECORDS = 10000
 N_MODES = 12
+FACTOR_SIZE = 0.8
 
 
-class CircularToyDataLoader(DataLoader):
+class LabeledCircularToyDataLoader(DataLoader):
     """
     A dataloader that returns samples from a simple toy problem 2d gaussian distributions of points in a circle
     """
 
     def __init__(self, use_batch=True, batch_size=100, n_batches=0, shuffle=False):
-        super().__init__(CircularToyDataSet, use_batch, batch_size, n_batches, shuffle)
+        super().__init__(LabeledCircularToyDataSet, use_batch, batch_size, n_batches, shuffle)
+
+    @property
+    def n_input_neurons(self):
+        return 2
+
+    @property
+    def num_classes(self):
+        return self.dataset().num_classes
+
+    def save_images(self, images, shape, filename):
+        self.dataset().save_images(images, filename)
+
+
+class UnlabeledCircularToyDataLoader(DataLoader):
```

Comment:
Seems very similar to `LabeledCirculrDataLoader` can it subclass it?

Reply:
Thanks. I have added subclasses for DataLoaders and DataSets.